whois-update/src/main/java/net/ripe/db/whois/update/domain/UpdateMessages.java

```diff
@@ -334,11 +335,10 @@ public static Message inetnumStatusLegacy() {
     }
 
     private static CharSequence intervalToString(final Interval<?> interval) {
-        if (interval instanceof Ipv4Resource) {
-            return ((Ipv4Resource) interval).toRangeString();
-        }
-
-        return interval.toString();
+        return switch (interval) {
+            case Ipv4Resource ipv4Resource -> ipv4Resource.toRangeString();
+            case Ipv6Resource ipv6Resource -> ipv6Resource.toString();
+        };
```

Comment:
Do you need null situation here as well?

Reply:
no because the original code was effectively an if-else and a null would have thrown an NPE